### PR TITLE
Opening a PR for Code review

### PR DESCRIPTION
**Pull request by tfhartmann**
Please give this code a review. I think this is about ready to rock but a second set of eyes would be awesome

### CODE REVIEW

**Comment by tfhartmann** [comment]
cc/ @hakamadare @pnealon

File: README.md [hunk 1]
```diff
@@ -1,2 +1,73 @@
...
+===========
+
 Terraform module for deploying and managing AWS Aurora
+
+This module
+
+- Stuff this module does
+
+- we encrypt everthing and make a kms id for you
+- future enhancment to make kms_key_id optional, so if you do pass it it'll use it, and if you don't it'll create the kms key id for you
```
Comment (hakamadare) [inline README.md]: `s/enhancment/enhancement/`

File: README.md [hunk 2]
```diff
@@ -1,2 +1,73 @@
...
+
+----------------------
+#### Required
+
+
+- `azs` - "A list of Availability Zones in the Region"
+- `cluster_size` - "Number of cluster instances to create"
+- `env` - "env to deploy into, should typically dev/staging/prod"
+- `name` - "Name for the Redis replication group i.e. cmsCommon"
+- `subnets` - "List of subnets to use in creating RDS subnet group (must already exist)"
```
Comment (hakamadare) [inline README.md]: specify whether it wants subnet IDs or CIDRs or ?

File: README.md [hunk 3]
```diff
@@ -1,2 +1,73 @@
...
+- `vpc_id`  - "VPC ID"
+
+##### see aws_rds_cluster documentation for these variables
```
Comment (hakamadare) [inline README.md]: for extra credit add a link to https://www.terraform.io/docs/providers/aws/r/rds_cluster.html

File: README.md [hunk 4]
```diff
@@ -1,2 +1,73 @@
...
+- `database_name`
+- `master_username`
+- `master_password`
+
+#### Optional
+- `backup_retention_period` - "The days to retain backups for defaults to 30"
+- `db_port` - "defaults to 3306"
+- `allowed_cidr` - "A list of Security Group ID's to allow access to. Defaults to ["127.0.0.1/32"]"
+- `allowed_security_groups` - "A list of Security Group ID's to allow access to. Defaults to empty list"
+- `preferred_backup_window` - "The daily time range during which automated backups are created. Default to 01:00-03:00"
```
Comment (hakamadare) [inline README.md]: specify that time is in UTC

File: README.md [hunk 5]
```diff
@@ -1,2 +1,73 @@
...
+- `instance_class` - "Instance class to use when creating RDS cluster. Defaults to db.r3.large"
```
Comment (hakamadare) [inline README.md]: db.t2.medium is a bit kinder to the person who is just kicking the tires on this module, or does that instance size bar us from doing multi-AZ?

File: README.md [hunk 6]
```diff
@@ -1,2 +1,73 @@
...
+  database_name           = "thtest"
+  master_username         = "tim"
+  master_password         = "test1234"
+  backup_retention_period = "1"
+  #preferred_backup_window = ""
+}
+
+```
+
+Outputs
```
Comment (hakamadare) [inline README.md]: we output reader and writer endpoints and security group ID, we should also output the ARN of the cluster

File: README.md [hunk 7]
```diff
@@ -1,2 +1,73 @@
...
+=======
+
+
+Authors
+=======
+
+[Tim Hartmann](https://github.com/tfhartmann)
+
+License
```
Comment (hakamadare) [inline README.md]: MIT license?

File: kms.tf [hunk 8]
```diff
@@ -0,0 +1,10 @@
+resource "aws_kms_key" "aurora" {
+  description = "RDS master key for ASICS Digital ${var.name}-${data.aws_vpc.vpc.tags["Name"]}"
```
Comment (hakamadare) [inline kms.tf]: remove reference to ASICS Digital :)

File: kms.tf [hunk 9]
```diff
@@ -0,0 +1,10 @@
...
+  deletion_window_in_days = 30
+  enable_key_rotation = "true"
+}
+
+resource "aws_kms_alias" "aurora" {
+  name = "alias/asics-digital-${var.name}-${data.aws_vpc.vpc.tags["Name"]}-rds-key"
```
Comment (hakamadare) [inline kms.tf]: also probably here?  however the kms alias has to be unique

File: main.tf [hunk 10]
```diff
@@ -0,0 +1,89 @@
+data "aws_vpc" "vpc" {
+  id = "${var.vpc_id}"
+}
+# ${data.aws_vpc.vpc.tags["Name"]}
```
Comment (hakamadare) [inline main.tf]: this commented line looks like a leftover

File: main.tf [hunk 11]
```diff
@@ -0,0 +1,89 @@
...
+
+resource "aws_rds_cluster" "aurora" {
+  cluster_identifier              = "tf-${var.name}-${data.aws_vpc.vpc.tags["Name"]}"
```
Comment (hakamadare) [inline main.tf]: this is AWESOME, i didn't know you could interpolate a hash like that!

File: main.tf [hunk 12]
```diff
@@ -0,0 +1,89 @@
...
+  subnet_ids = ["${var.subnets}"]
+
+  tags {
+    Name = "tf-rds-${var.name}-${data.aws_vpc.vpc.tags["Name"]}"
+  }
+}
+
+resource "aws_db_parameter_group" "aurora_parameter_group" {
+  name        = "tf-rds-${var.name}-${data.aws_vpc.vpc.tags["Name"]}"
+  family      = "aurora5.6"
```
Comment (hakamadare) [inline main.tf]: at some point we'll probably want to make this a parameter, don't need it now however

File: main.tf [hunk 13]
```diff
@@ -0,0 +1,89 @@
...
+  description = "Terraform-managed parameter group for ${var.name}-${data.aws_vpc.vpc.tags["Name"]}"
+
+  tags {
+    Name = "tf-rds-${var.name}-${data.aws_vpc.vpc.tags["Name"]}"
+  }
+}
+
+resource "aws_rds_cluster_parameter_group" "aurora_cluster_parameter_group" {
+  name        = "tf-rds-${var.name}-${data.aws_vpc.vpc.tags["Name"]}"
+  family      = "aurora5.6" # does this need a var?
```
Comment (hakamadare) [inline main.tf]: see previous comment :)

File: main.tf [hunk 14]
```diff
@@ -0,0 +1,89 @@
...
+  tags {
+    Name = "tf-rds-${var.name}-${data.aws_vpc.vpc.tags["Name"]}"
+  }
+}
+
+resource "aws_db_option_group" "aurora_option_group" {
+  name                     = "tf-rds-${var.name}-${data.aws_vpc.vpc.tags["Name"]}"
+  option_group_description = "Terraform-managed option group for ${var.name}-${data.aws_vpc.vpc.tags["Name"]}"
+  engine_name              = "aurora"
+  major_engine_version     = "5.6"
```
Comment (hakamadare) [inline main.tf]: will also need to be parameterized once there are multiple valid values for this option

File: security_group.tf [hunk 15]
```diff
@@ -0,0 +1,28 @@
+resource "aws_security_group" "aurora_security_group" {
+  name        = "tf-sg-rds-${var.name}-${data.aws_vpc.vpc.tags["Name"]}"
+  description = "Terraform-managed RDS security group for ${var.name}-${data.aws_vpc.vpc.tags["Name"]}"
+  vpc_id      = "${var.vpc_id}"
```
Comment (hakamadare) [inline security_group.tf]: this could be `${data.aws_vpc.vpc.id}`, same with other instances of `${var.vpc_id}`

File: README.md (hunk 5 above)
Comment (tfhartmann) [inline README.md]: It doesn't seem to. Just tested it, and it seems to be modifying the cluster OK.  This is a good suggestion, I'll make this change!

File: kms.tf (hunk 8 above)
Comment (tfhartmann) [inline kms.tf]: 😢  whoops

File: kms.tf (hunk 9 above)
Comment (tfhartmann) [inline kms.tf]: so I think that `alias/${var.name}-${data.aws_vpc.vpc.tags["Name"]}-rds-key` should give us enough uniqueness actually. I'll remove the `asics-digital-` prefix.

File: main.tf (hunk 10 above)
Comment (tfhartmann) [inline main.tf]: indeed it is!

File: main.tf (hunk 12 above)
Comment (tfhartmann) [inline main.tf]: Yeah, I did some hand wringing over that too, I err'd on the side of leaving it, so as not to gild the lilly to much...  it stuck out at me too

File: main.tf (hunk 13 above)
Comment (tfhartmann) [inline main.tf]: 👍

File: main.tf (hunk 14 above)
Comment (tfhartmann) [inline main.tf]: /agree

File: security_group.tf (hunk 15 above)
Comment (tfhartmann) [inline security_group.tf]: Yeah we could use the output of the data source rather then the direct var, that would let us change up the input to the data source if we ever wanted to.  The only other use of `${var.vpc_id}` in the module is here https://github.com/FitnessKeeper/terraform-aurora/blob/b760ad1ccbd88bf534b79c07f921c78c61f52d5b/main.tf#L2  Which we can't really swap out, as it's needed to initialize the aws_vpc data source.  Or was there one somewhere else that I missed?

File: README.md (hunk 7 above)
Comment (tfhartmann) [inline README.md]: updated in 2e00d44 because Markdown defeats me apparently...

**Pull request merged**